### PR TITLE
R4R: Setup different level min-deposit for different proposals

### PR DESCRIPTION
**Pull request by MrXJC**

### CODE REVIEW

File: types/gov/params/gov_params.go [hunk 1]
```diff
@@ -101,20 +127,29 @@ func (param *DepositProcedureParam) Valid(jsonStr string) sdk.Error {
...
+			return sdk.NewError(params.DefaultCodespace, params.CodeInvalidMinDepositAmount, fmt.Sprintf(IMPORTANT+"MinDepositAmount"+param.Value.CriticalMinDeposit[0].String()+" should be larger than 10iris and less than 10000iris"))
+		}
 
+		if param.Value.NormalMinDeposit[0].Amount.LT(LowerBound.Amount) || param.Value.NormalMinDeposit[0].Amount.GT(UpperBound.Amount) {
+			return sdk.NewError(params.DefaultCodespace, params.CodeInvalidMinDepositAmount, fmt.Sprintf(NORMAL+"MinDepositAmount"+param.Value.CriticalMinDeposit[0].String()+" should be larger than 10iris and less than 10000iris"))
 		}
 
 		if param.Value.MaxDepositPeriod.Seconds() < 20 || param.Value.MaxDepositPeriod.Seconds() > THREE_DAYS {
-			return sdk.NewError(params.DefaultCodespace, params.CodeInvalidDepositPeriod, fmt.Sprintf("MaxDepositPeriod (%s) should be larger than 20s and less than %ds",strconv.Itoa(int(param.Value.MaxDepositPeriod.Seconds())), THREE_DAYS))
+			return sdk.NewError(params.DefaultCodespace, params.CodeInvalidDepositPeriod, fmt.Sprintf("MaxDepositPeriod (%s) should be larger than 20s and less than %ds", strconv.Itoa(int(param.Value.MaxDepositPeriod.Seconds())), THREE_DAYS))
```
Comment (chengwenxi) [inline types/gov/params/gov_params.go]: if period ==20 ||  period ==THREE_DAYS?

File: types/gov/params/gov_params.go [hunk 2]
```diff
@@ -27,15 +36,29 @@ type ParamSet struct {
...
+func NewDepositProcedure() DepositProcedure {
+	var ciriticalMinDeposit, _ = types.NewDefaultCoinType(stakeTypes.StakeDenomName).ConvertToMinCoin(fmt.Sprintf("%d%s", CRITICAL_DEPOSIT, stakeTypes.StakeDenomName))
+	var importantMinDeposit, _ = types.NewDefaultCoinType(stakeTypes.StakeDenomName).ConvertToMinCoin(fmt.Sprintf("%d%s", IMPORTANT_DEPOSIT, stakeTypes.StakeDenomName))
+	var normalMinDeposit, _ = types.NewDefaultCoinType(stakeTypes.StakeDenomName).ConvertToMinCoin(fmt.Sprintf("%d%s", NORMAL_DEPOSIT, stakeTypes.StakeDenomName))
+
+	return DepositProcedure{
+		CriticalMinDeposit:  sdk.Coins{ciriticalMinDeposit},
+		ImportantMinDeposit: sdk.Coins{importantMinDeposit},
+		NormalMinDeposit:    sdk.Coins{normalMinDeposit},
+		MaxDepositPeriod:    time.Duration(172800) * time.Second}
```
Comment (wukongcheng) [inline types/gov/params/gov_params.go]: magic number to const

File: types/gov/proposals.go [hunk 3]
```diff
@@ -362,3 +362,34 @@ func (resultA TallyResult) Equals(resultB TallyResult) bool {
...
+	case ProposalTypeParameterChange:
+		return ProposalLevelImportant
+	case ProposalTypeSoftwareHalt:
+		return ProposalLevelCritical
+	case ProposalTypeSoftwareUpgrade:
+		return ProposalLevelCritical
+	default:
+		return  ProposalLevelNil
+	}
+}
```
Comment (chengwenxi) [inline types/gov/proposals.go]: fmt and function rename to (p Proposal) Level() ProposalLevel will better?

File: modules/gov/govparams_util.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	govparams.DepositProcedureParameter.LoadValue(ctx)
+	switch GetProposalLevel(p) {
+	case ProposalLevelCritical:
+		return govparams.DepositProcedureParameter.Value.CriticalMinDeposit
+	case ProposalLevelImportant:
+		return govparams.DepositProcedureParameter.Value.ImportantMinDeposit
+	case ProposalLevelNormal:
+		return govparams.DepositProcedureParameter.Value.NormalMinDeposit
+	default:
+		return govparams.DepositProcedureParameter.Value.NormalMinDeposit
```
Comment (wukongcheng) [inline modules/gov/govparams_util.go]: So ProposalLevelNil == NormalMinDeposit due to the default case ?

File: modules/gov/simulation/msgs.go [hunk 5]
```diff
@@ -126,6 +126,9 @@ func simulationCreateMsgSubmitProposal(r *rand.Rand, sender simulation.Account)
 		sender.Address,
 		deposit,
 		param,
+		uint64(0),
+		" ",
+		1,
```
Comment (wukongcheng) [inline modules/gov/simulation/msgs.go]: mistake to add 3 more params

**Pull request merged**